apps/competitions-api/src/routes/competitions/createCompetition.ts

```diff
@@ -1,29 +1,19 @@
 import { Router } from 'express';
 import { prisma } from '@competition-manager/prisma';
-import { z } from 'zod';
 import { parseRequest, AuthenticatedRequest, checkRole } from '@competition-manager/utils';
-import { Competition$, BaseCompetition$ } from '@competition-manager/schemas';
+import { BaseCompetitionWithRelationId$ ,BaseCompetition$ } from '@competition-manager/schemas';
 
 export const router = Router();
 
-const Body$ = Competition$.pick({
-    name: true,
-    date: true,
-    method: true
-}).extend({
-    paymentPlanId: z.number(),
-    optionsId: z.array(z.number()).optional(),
-});
-
 router.post(
     '/',
-    parseRequest('body', Body$),
+    parseRequest('body', BaseCompetitionWithRelationId$),
     checkRole('club'),
     async (req: AuthenticatedRequest, res) => {
         // TODO: stripe
 
 
-        const body = Body$.parse(req.body);
+        const body = BaseCompetitionWithRelationId$.parse(req.body);
         const competition = BaseCompetition$.parse(body);
         const { paymentPlanId, optionsId } = body;
```

Comment:
A tester je me demande si on ne peux pas faire un truc comme ceci
```suggestion
        const body = BaseCompetitionWithRelationId$.parse(req.body);
        const competition = BaseCompetition$.parse(body);
        const { paymentPlanId, optionsId } = body;
```
```suggestion
        const { paymentPlanId, optionsId, ...competitionData } = BaseCompetitionWithRelationId$.parse(req.body);
        const competition = BaseCompetition$.parse(competitionData );
```
Pas sûr du tout mais à tester pcq si ça marche je trouve ça mieux